llap-server/src/test/org/apache/hadoop/hive/llap/cache/TestCacheAllocationsEvictionsCycles.java

```diff
@@ -0,0 +1,324 @@
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.hive.llap.cache;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hive.common.io.Allocator;
+import org.apache.hadoop.hive.common.io.encoded.MemoryBuffer;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.llap.io.metadata.MetadataCache;
+import org.apache.hadoop.hive.llap.metrics.LlapDaemonCacheMetrics;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Test the full cycle of allocation/accounting/eviction interactions.
+ */
+public class TestCacheAllocationsEvictionsCycles {
+
+  private static final Logger LOG = LoggerFactory.getLogger(TestCacheAllocationsEvictionsCycles.class);
+  private static final LlapDaemonCacheMetrics cacheMetrics = LlapDaemonCacheMetrics.create("testCache", "testSession");
+
+  private final long maxSize = 1024;
+  private final LowLevelCache dataCache = Mockito.mock(LowLevelCache.class);
+  private final SerDeLowLevelCacheImpl serdCache = Mockito.mock(SerDeLowLevelCacheImpl.class);
+  private final MetadataCache metaDataCache = Mockito.mock(MetadataCache.class);
+
+  private BuddyAllocator allocator;
+  private MemoryManager memoryManager;
+  private LowLevelCachePolicy cachePolicy;
+  private EvictionTracker evictionTracker;
+
+  @Before public void setUp() throws Exception {
+    Configuration conf = new Configuration();
+    // Set lambda to 1 so the heap size becomes 1 (LRU).
+    conf.setDouble(HiveConf.ConfVars.LLAP_LRFU_LAMBDA.varname, 1.0f);
+    int minBufferSize = 1;
+    cachePolicy = new LowLevelLrfuCachePolicy(minBufferSize, maxSize, conf);
+    memoryManager = new LowLevelCacheMemoryManager(maxSize, cachePolicy, cacheMetrics);
+    int maxAllocationSize = 1024;
+    int minAllocationSize = 8;
+    allocator =
+        new BuddyAllocator(true,
+            false,
+            minAllocationSize,
+            maxAllocationSize,
+            1,
+            maxSize,
+            0,
+            null,
+            memoryManager,
+            cacheMetrics,
+            "no-force-eviction",
+            true, 1024 * 1024 * 128);
+    EvictionDispatcher evictionDispatcher = new EvictionDispatcher(dataCache, serdCache, metaDataCache, allocator);
+    evictionTracker = new EvictionTracker(evictionDispatcher);
+    cachePolicy.setEvictionListener(evictionTracker);
+  }
+
+  @After public void tearDown() throws Exception {
+    LOG.info("Purge the cache on tear down");
+    cachePolicy.purge();
+    allocator = null;
+    memoryManager = null;
+    cachePolicy = null;
+  }
+
+  /**
+   * Test case to ensure that deallocate it does merge small blocks into bigger ones.
+   */
+  @Test(timeout = 6_000L) public void TestMergeOfBlocksAfterDeallocate() {
```

Comment:
Seems, that all other tests (that I read so far) in LLAP followed the standard Java method name convention, to start with a lowercase t.

Reply:
fixed.